core/src/main/scala/org/apache/spark/h2o/converters/WriteConverterCtxUtils.scala

```diff
@@ -112,6 +112,22 @@ object WriteConverterCtxUtils {
           Log.debug(node + ": " + n + " chunks.")
       }
     }
+
+    val layouts = fr.vecs().map(_.espc())
+
+    if(!layouts.isEmpty){
```

Comment:
What if ESPC is correct but the chunks itself are broken? Eg. there is a chunk that has less data than what the ESPC states. I think that might be the true cause of the bug

Reply:
I see, good point. I will check also real shape of data